From 79b78e53723805530edd85a2696965d160afd85b Mon Sep 17 00:00:00 2001
From: Quanyang Wang <quanyang.wang@windriver.com>
Date: Sun, 6 Jul 2025 10:56:52 +0800
Subject: [PATCH] fix implicit-function-declaration

This patch is to fix the following building error:
	| microblazeel-v10.0-bs-cmp-re-mh-div-xilinx-elf/libgloss/
	4.4.0+git/git/libgloss/microblaze/linux-inbyte.c:4:9: error: implicit declaration of function 'read' [-Wimplicit-function-declaration]
	|     4 |         read(0, &ch, 1);
	|       |         ^~~~

Upstream-Status: Pending

Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 libgloss/microblaze/linux-inbyte.c  | 2 ++
 libgloss/microblaze/linux-outbyte.c | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/libgloss/microblaze/linux-inbyte.c b/libgloss/microblaze/linux-inbyte.c
index 58fdf19..98872e1 100644
--- a/libgloss/microblaze/linux-inbyte.c
+++ b/libgloss/microblaze/linux-inbyte.c
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 int inbyte(void)
 {
 	char ch = 0;
diff --git a/libgloss/microblaze/linux-outbyte.c b/libgloss/microblaze/linux-outbyte.c
index 9d7bc48..50c4532 100644
--- a/libgloss/microblaze/linux-outbyte.c
+++ b/libgloss/microblaze/linux-outbyte.c
@@ -1,3 +1,7 @@
+#include <unistd.h>
+
+extern ssize_t _write(int fd, const void *b, size_t count);
+
 void outbyte (unsigned char c)
 {
 	_write(1, &c, 1);
-- 
2.40.0

